pkg/workflow/client.go

```diff
@@ -304,9 +313,32 @@ func (s *ArgoWorkflowClient) Terminate(ctx context.Context, opts TerminateOption
 		return err
 	}
 
+	if err == nil {
+		return nil
+	}
+
+	if err := s.setNotFoundStatus(ctx, opts.WorkflowExecution); err != nil {
```

Comment:
Why do we handle not-found differently comparing to the Terminate() error?

Reply:
When a workflow is not found in Kubernetes, the status of the workflow remains unchanged even when a user attempts to stop it, resulting in a perpetual 'running' status. We require a method to address this situation.

The reason we don't need to handle a `not found` error in `resubmit`, `submit` or `delete` is that the absence of a workflow in Kubernetes does not impact the outcome.

Reply:
What about we update WorkflowExecutionStatusRunning when handling Terminate() error?

Reply:
Updated.